difference-of-squares/test_difference-of-squares.sml

```diff
@@ -0,0 +1,83 @@
+use "difference-of-squares.sml";
```

Comment:
This is the current tree for the exercises:

```
$ tree exercises/
exercises/
├── accumulate
│   ├── example.sml
│   └── test_accumulate.sml
├── allergies
│   ├── example.sml
│   └── test_allergies.sml
├── anagram
│   ├── example.sml
│   └── test_anagram.sml
├── binary
│   ├── example.sml
│   └── test_binary.sml
├── flatten-array
│   ├── example.sml
│   └── test_flatten_array.sml
├── hamming
│   ├── example.sml
│   └── test_hamming.sml
└── nth-prime
    ├── example.sml
    └── test_nth_prime.sml
```

So it looks like we're using underscores, not hyphens in the file names.